langchain/src/vectorstores/similarity-score-threshold.ts

```diff
@@ -0,0 +1,72 @@
+import { VectorStore, VectorStoreRetriever } from "./base.js";
+import { Document } from "../document.js";
+
+export type SimilarityScoreFilters = {
+  minSimilarityScore: number;
+  dynamicK?: boolean;
+  kIncrement?: number;
+  maxK?: number;
+};
+
+export class SimilarityScoreThresholdVectorStoreRetriever {
+  minSimilarityScore: number;
+
+  dynamicK = false;
+
+  kIncrement = 1;
+
+  maxK = 100;
+
+  constructor(similarityScoreFilters: SimilarityScoreFilters) {
+    this.minSimilarityScore = similarityScoreFilters.minSimilarityScore;
+    this.dynamicK = similarityScoreFilters.dynamicK ?? this.dynamicK;
+    this.kIncrement = similarityScoreFilters.kIncrement ?? this.kIncrement;
+    this.maxK = similarityScoreFilters.maxK ?? this.maxK;
+  }
+
+  private isAboveMinScore([, score]: [Document, number]): boolean {
+    return score >= this.minSimilarityScore;
+  }
+
+  fromVectorStore(vectorStoreRetriever: VectorStoreRetriever<VectorStore>) {
```

Comment:
This can just take a `VectorStore` and call `.asRetriever()`.

Reply:
I don't think so. Users can call `.asRetriever()` providing the initial `k ` value and the vector store filters in an object.

Do you think we should receive these props and pass them to it, or should we keep it as is?